collector/src/execute.rs

```diff
@@ -131,7 +131,11 @@ struct BenchmarkConfig {
     #[serde(default)]
     touch_file: Option<String>,
 
-    category: Category,
+    /// If this field is `None` when read from `perf-config.json`:
+    /// - If `supports_stable` is true, it will be replaced with
+    ///   `Some(Primary)` when the result is recorded in the DB.
+    /// - If `supports_stable` is false, an error will be returned.
+    category: Option<Category>,
```

Comment:
Maybe we should have a StableOnly category or something like that? (Potentially in the long run fully replacing supports_stable with that category).